### ISSUE TITLE
Navigation page: the "Currently editing" select element should not trigger a page re-render on change

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#25312]
**Describe the bug**

In the new Navigation page, the "Currently editing" select element triggers a page re-render when the change event fires. This is a known accessibility barrier as changes of context should always happen on explicit user activation. Worth reminding this behavior is dangerous especially on Windows, where the select elements options can be changed even _keeping the select closed_. For more details, please refer to this Trac ticket:

**Review the usage of the change event on select elements**
https://core.trac.wordpress.org/ticket/40925

The WCAG guidelines explicitly state to provide a submit button to initiate a change of context.

Also, worth noting the current Menus page does have a button to submit the select element changed value, and for good reasons:

<img width="547" alt="Screenshot 2020-09-14 at 19 28 47" src="https://user-images.githubusercontent.com/1682452/93118439-9f995000-f6c0-11ea-8666-b1008de51b92.png">

**To reproduce**

Animated GIF of the current behavior:

![navigation select new](https://user-images.githubusercontent.com/1682452/93118072-00745880-f6c0-11ea-96f1-d9743efc26ca.gif)

- on Windows
- tab to the Select element
- use the down arrow key to change value 
- as soon as the value changes, the page re-renders showing the new menu even if the user hasn't explicitly submitted the change
- this is even more harming because focus is then moved to the menu being edited: there's a specific issue for that, see #25298 

**Expected behavior**
The page to not re-render on the change event. There should be a button to initiate a change of context.

**Editor version (please complete the following information):**
- WordPress version: 5.6 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? plugin
- If the Gutenberg plugin is installed, which version is it? 9.0.0-rc.1

**Comment by shaunandrews** [comment]
I'm thinking it might be better to consolidate the interactions; All the other buttons here (Add new, Manage locations) are `DropdownMenu`'s and maybe this one should be too. It would look something like this:

![dropdownmenu](https://user-images.githubusercontent.com/191598/93532540-352a1f00-f90f-11ea-9e94-f8b63e9c35b7.gif)

**Comment by afercia** [comment]
Yep a `DropdownMenu` is an option and I do see it brings in consistency. However, it doesn't show which is the currently selected menu being edited. This missing info could be communicated separately though.

Given this page needs also some more headings to better identify the main sections (worth reminding headings are also a navigational tool for screen reader users) I'd like to propose to add a `<h2>` heading before the menu toolbar and the heading text would be the name of the menu being edited.

Aside: the tre view on the right has a `<h3>` heading with text "List view" that should be bumped up to a `<h2>` as the headings hierarchy is skipping one level.

I see @talldan made a PR for a select + button in the meantime. Going to have a look but I like the idea of a `DropdownMenu`.

**Comment by shaunandrews** [comment]
> Aside: the tree view on the right has a <h3> heading with text "List view" that should be bumped up to a <h2> as the headings hierarchy is skipping one level.

If we do add a <h2> heading for the current menu indicator, wouldn't it make sense for the list view heading to remain an <h3>?

**Comment by afercia** [comment]
Hm... not sure. Do you see the "list view" as a sub-section of the menu-edit section? Or are they two sections of the same importance? Generally, fewer heading levels are better because are simpler and faster to navigate through. I'd go for a H3 only for a section that is clearly, logically a sub-section.

This has nothing to do with the visual side: headings don't necessarily need to look "big", as long as the visual hierarchy is preserved besides the semantic hierarchy.
